### ISSUE TITLE
box width with breakpoints doesn't work for first entry

### ISSUE THREAD

**Issue by publicJorn** [rebassjs/rebass#793]
Hey, want to implement reponsive layout. I couldn't get this to work, even when copying the exact code from the docs.

Here is the code: https://codesandbox.io/s/box-width-with-breakpoints-doesnt-seem-to-work-5r6io

I copied it literally from the [theming](https://rebassjs.org/reflexbox/#theming) and [responsive styles](https://rebassjs.org/reflexbox/#responsive-styles) part of the docs for reflexbox.

You can see that the first breakpoint (100% width at 40em) doesn't work. The other two (50% at 52em and 25% at 64em) do work.

Are the docs outdated, or am I still doing something wrong? please let me know!

**Comment by publicJorn** [comment]
BTW: this used to work in the old `@rebass/grid` package. You can see it in action in this sandbox: https://codesandbox.io/s/reflexbox-width-with-styled-components-50pjg

`reflexbox` is also a dependency here, so you can easily switch between the two and see what works and what doesn't.

**Comment by xeoneux** [comment]
It works for me when I use the sandbox link you posted

**Comment by publicJorn** [comment]
@xeoneux which sandbox works for you?

For me the first one doesn't work for the first breakpoint (100%). Try uncommenting the `Flex` and the second `Box` in the code. This makes it even more clear that it doesn't fill out to 100%.
 
The second one works, but only for the deprecated `@rebass/grid` package. If you comment that out and uncomment `reflexbox` it doesn't.

**Comment by publicJorn** [comment]
I've investigated more. It seems that in the default example the first (100%) width is just too wide. So in the split-screen preview window of codesandbox it never matches. If I put the first value to `0em` instead of `44em` it works.

However, the second example still doesn't work. I narrowed it down to being an issue with the ThemeProvider. It works for emotion, but not for styled-components. I'll make a new issue for that:  #798
